bootstrap/README.md

```diff
@@ -1,14 +1,161 @@
+                    Terraform changes (default: terraform-iac-pipeline).
+      -b (optional) Bucket name without 'gs://' which will store the Terraform
+                    state files (default: 'tf-state-<GCP_PROJECT_ID>').
+      -o (optional) relative or absolute path to directory that will store the
+                    generated Terraform code (default: 'iac-output').
+      -g (optional) GitHub repository in the format '<owner/org>/<reponame>'. If
+                    given, the Terraform code will be configured to enable the Workload
+                    Identity Federation support for GitHub Workflows. This is
+                    required for keyless authentication from GitHub Workflows which
+                    is strongly recommended. This can also be set up later.
+      -t (optional) Time to sleep for in between bootstrap stages exectution,
+              required for GCP IAM changes to propagate (default: '5m').
+      -n (optional) If set, no Terraform code is generated. Only the service
+                    account and state bucket are created. In addition the needed
+                    APIs are enabled. Only use this option if you use a different
+                    template for newly created GCP projects.
 ```
+
+## How it works
+
+The `bootstrap.sh` script is using [Terraform](https://www.terraform.io/) to execute set of actions on a given [GCP project](https://cloud.google.com/resource-manager/docs/creating-managing-projects) in order to prepare it for further usage with Terraform and [Service Account impersonation](https://cloud.google.com/docs/authentication/use-service-account-impersonation) in a standardised way, as expected and recommended by the Cloud Foundation.
+
+The whole bootstrapping process is happening in two stages:
+
+1. Terraform code from the `terraform` directory is being run using the executing user access to the [GCP IAM](https://cloud.google.com/iam/docs/groups-in-cloud-console) `manager_group` and does the following (in a nutshell):
+
+   - Grants the `manager_group` additional roles needed to perform the whole process
+   - Creates a service account intended for future use with Terraform IaC (also used in second stage)
+   - Creates a [GCP storage bucket](https://cloud.google.com/storage/docs/buckets) for Terraform [state file](https://developer.hashicorp.com/terraform/language/settings/backends/gcs)
+   - Generates Terraform code in output directory (by default: `iac-output`) to be used in second stage and further by the project users
+   - Sleeps for a given duration (by default: 5m) to give GCP time to [synchronize the IAM changes](https://cloud.google.com/iam/docs/access-change-propagation) to the groups, service accounts, etc.
+
+1. Terraform code from the `iac-output` (or a different one, if the `-o` parameter was used) is being run using the service account created in the first stage and does the following (in a nutshell):
```

Comment:
a small typo: 1. -> 2.

Reply:
This isn't a typo, right? This is Markdown syntax for the ordered lists. Can you check the document rendering with Markdown and confirm it's not a typo please?

Reply:
yes you're correct.